fem/nonlinearform.cpp

```diff
@@ -24,7 +24,7 @@ void NonlinearForm::SetAssemblyLevel(AssemblyLevel assembly_level)
    switch (assembly)
    {
       case AssemblyLevel::NONE:
-         // This is the default behavior.
+         ext = new MFNonlinearFormExtension(this);
```

Comment:
Currently, in the default constructor `AssemblyLevel::NONE` is used as the default. We need to change that to something different since now we want `AssemblyLevel::NONE` to mean the new "matrix-free" (`MF` is not the best prefix for nonlinear operator since there is no "matrix" version anyway) assembly level.

We can either introduce `AssemblyLevel::LEGACYNONE` or rename `AssemblyLevel::LEGACYFULL` to `AssemblyLevel::LEGACY` and use that to denote the "legacy" assembly level for both `BilinearForm` and `NonlinearForm`.

What do you think?

Reply:
I was proposing both solutions to @tzanio , I'm fine with either, with a slight preference for `AssemblyLevel::LEGACY`.

Reply:
`AssemblyLevel::LEGACY` is ok with me

Reply:
For backward compatibility, I think it maybe good to keep the constant `LEGACYFULL = 0` in addition to `LEGACY = 0`.